src/lib.rs

```diff
@@ -1,5 +1,187 @@
+    clippy::overly_complex_bool_expr, // error by default
+    clippy::option_map_unit_fn,
+    clippy::partialeq_to_none,
+    clippy::precedence,
+    clippy::redundant_closure,
+    clippy::redundant_pattern_matching,
+    clippy::redundant_static_lifetimes,
+    clippy::search_is_some,
+    clippy::too_many_arguments,
+    clippy::type_complexity,
+    clippy::unit_arg,
+    clippy::uninlined_format_args,
+    clippy::unnecessary_cast,
+    clippy::unnecessary_fallible_conversions,
+    clippy::unnecessary_map_on_constructor,
+    clippy::unnecessary_mut_passed,
+    clippy::unnecessary_lazy_evaluations,
+    clippy::unneeded_wildcard_pattern,
+    clippy::upper_case_acronyms,
+    clippy::useless_conversion,
+    clippy::zero_prefixed_literal,
+)]
+#![deny(
+    unsafe_op_in_unsafe_fn,
+    clippy::missing_safety_doc,
+    clippy::undocumented_unsafe_blocks
+)]
 
+#[cfg(not(any(feature = "bitdepth_8", feature = "bitdepth_16")))]
+compile_error!("No bitdepths enabled. Enable one or more of the following features: `bitdepth_8`, `bitdepth_16`");
+
+use parking_lot::Mutex;
+use std::cmp;
+use std::ffi::c_char;
+use std::ffi::c_uint;
+use std::ffi::c_void;
+use std::ffi::CStr;
+use std::mem;
+use std::ptr;
+use std::ptr::NonNull;
+use std::slice;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::AtomicU32;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use std::sync::Once;
+use std::thread;
+use to_method::To as _;
+
+pub mod include {
+    pub mod common {
+        pub(crate) mod attributes;
+        pub(crate) mod bitdepth;
+        pub(crate) mod dump;
+        pub(crate) mod intops;
+        pub(crate) mod validate;
+    } // mod common
+    pub mod dav1d {
+        pub mod common;
+        pub mod data;
+        pub mod dav1d;
+        pub mod headers;
+        pub mod picture;
+    } // mod dav1d
+} // mod include
+
+pub mod align;
+pub(crate) mod assume;
+pub(crate) mod c_arc;
+pub(crate) mod c_box;
+mod cdef;
+mod cdef_apply;
+mod cdf;
+mod const_fn;
+pub mod cpu;
+mod ctx;
+mod cursor;
+mod data;
+mod decode;
+mod dequant_tables;
+pub(crate) mod disjoint_mut;
+pub(crate) mod enum_map;
+mod env;
+pub(crate) mod error;
+mod ffi_safe;
+mod fg_apply;
+mod filmgrain;
+mod getbits;
+pub(crate) mod pic_or_buf;
+pub(crate) mod pixels;
+pub(crate) mod relaxed_atomic;
+pub mod send_sync_non_null;
+pub(crate) mod strided;
+pub(crate) mod with_offset;
+pub(crate) mod wrap_fn_ptr;
+// TODO(kkysen) Temporarily `pub(crate)` due to a `pub use` until TAIT.
+mod extensions;
+mod in_range;
+mod internal;
+mod intra_edge;
+mod ipred;
+mod ipred_prepare;
+mod iter;
+mod itx;
+mod itx_1d;
+pub(crate) mod levels;
+mod lf_apply;
+mod lf_mask;
+pub(crate) mod log;
+mod loopfilter;
+mod looprestoration;
+mod lr_apply;
+mod mc;
+#[path = "mem.rs"]
+mod memmod;
+mod msac;
+mod obu;
+mod pal;
+mod picture;
+mod qm;
+mod recon;
+mod refmvs;
+mod scan;
+mod tables;
+mod thread_task;
+mod warpmv;
+mod wedge;
+
+pub use error::Dav1dResult;
```

Comment:
```suggestion
pub use crate::error::Dav1dResult;
```

We've been trying to be consistent about using `crate::` so there's less ambiguity.

Reply:
Although I think `lib.rs` previously lacked a `crate::` here, too.

Reply:
I don't understand the ambiguity issue. It's not like there is an `error` crate, nor are there multiple `error` modules? It seems more idiomatic to me import directly from the module that is declared in the same scope.

Reply:
I don't mean `error` specifically is ambiguous, but for example, `mem` is, and referring to the module directly without going through `crate::` means that it'd change if you moved it to another module.